### DISCUSSION TITLE
Chart types that you'd like to see in Lightdash

### DISCUSSION THREAD

**Discussion by TuringLovesDeathMetal** [lightdash/lightdash#1175] [Ideas]
Charts are a pretty important part of Lightdash, and we know we're missing a few chart types in the app.

We'd love to hear it from you: 

### what chart type would you like to see added to Lightdash? 📈 

(Be sure to check the comments below and upvote if someone has already added your chart type!)

**Comment by TuringLovesDeathMetal** [comment]
"Big Number" type visualization

(already a ticket for this [here](https://github.com/lightdash/lightdash/issues/940))

**Reply by tomcent-tom** [reply]
Similarly, a gauge chart would be nice to have.
Not crucial. Just nice to have.

**Reply by TuringLovesDeathMetal** [reply]
oooh! I've never actually used a gauge chart before - love it.

It's like being in Formula 1, but actually you're in an exec meeting talking about ARR, CTR, WAUs and any other fun acronyms you can think of 🙂

**Reply by TuringLovesDeathMetal** [reply]
V1 of this is complete! #1264

**Reply by risafangs** [reply]
I love this type of chart with a % change from previous period. I also wish I could choose which column you take the number from - it seems like table calculations don't work even if I move them to come first.

**Reply by TuringLovesDeathMetal** [reply]
@risafangs totally. We have an open issue for that, actually: https://github.com/lightdash/lightdash/issues/1520

We're working on some chart configurations in the next two weeks, so I'll make sure to add it to the backlog :)

**Comment by TuringLovesDeathMetal** [comment]
Table 
(already a ticket for this [here](https://github.com/lightdash/lightdash/issues/430))

**Reply by TuringLovesDeathMetal** [reply]
V1 of this is complete! #430

**Comment by TuringLovesDeathMetal** [comment]
Pie chart

**Reply by clausherther** [reply]
![image](https://user-images.githubusercontent.com/7342749/150182487-365b3cb8-42d3-44f4-82ec-ee5cd04983b5.png)

**Comment by TuringLovesDeathMetal** [comment]
Map

**Reply by olly-writes-code** [reply]
Would love this! Consider [deck.gl](https://deck.gl/) for performant maps when rendering lots of data

**Reply by TuringLovesDeathMetal** [reply]
oooh @olly-writes-code and it's open source 👀 love it! We use eCharts as our charting library..I wonder if deck.gl plays nicely with it 🤞 (from our experience, e-charts can be a bit of a bully to integrations 😅 so I'll let the engineering team take a better look at it).

Linked in the open issue here! #3015

**Comment by TuringLovesDeathMetal** [comment]
Stacked bar charts (both % stacked + raw value stacked)

(already a ticket for this [here](https://github.com/lightdash/lightdash/issues/820))

**Comment by djayatillake** [comment]
Waterfall 😈

**Reply by djayatillake** [reply]
It's really useful but annoying to  implement unless your graphing library supports it really well.

**Reply by TuringLovesDeathMetal** [reply]
oh boy....can't wait to try this one out

**Comment by clausherther** [comment]
Combination two-axis charts, e.g vertical bar (axis 1) + line (axis 2)

**Reply by owlas** [reply]
Great suggestion @clausherther ! This is essential

**Reply by pinkbaka** [reply]
bump

**Reply by tanguy-nsd** [reply]
+1

**Comment by clausherther** [comment]
~Vertical~ Horizontal bar chart

**Reply by TuringLovesDeathMetal** [reply]
This already exists (unless I'm misunderstanding something here 🤔 )

Is it something different to this? 
![image](https://user-images.githubusercontent.com/31848148/150957305-75075699-afe6-4f40-9577-c845dc90dc4b.png)

**Reply by clausherther** [reply]
Hi @TuringLovesDeathMetal ! I think I had the wrong term for this, turns out it's called a Horizontal Bar chart (I guess the x-axis is what makes this horizontal, not the bars) ;) 
![image](https://user-images.githubusercontent.com/7342749/150993591-984f6da8-5839-4c93-b558-394a4fd9f709.png)

**Reply by TuringLovesDeathMetal** [reply]
it's your lucky day - this one exists too! I hope it feels like we're making all your bar chart dreams come true 🪄 🤗 

![image](https://user-images.githubusercontent.com/31848148/151203096-a1e92af8-966a-4f75-a3cf-344344adee60.png)

**Reply by clausherther** [reply]
I'm so impressed! :)

**Reply by TuringLovesDeathMetal** [reply]
That's what we like to hear!

**Comment by TuringLovesDeathMetal** [comment]
Histogram

**Comment by TuringLovesDeathMetal** [comment]
Funnel

**Comment by owlas** [comment]
Gauge chart 
<img width="804" alt="Screenshot 2022-01-27 at 22 52 42" src="https://user-images.githubusercontent.com/11660098/151456572-8384076b-af2c-4634-8f4c-780d5b16e490.png">

From @tomcent-tom

**Reply by owlas** [reply]
Ideally you should be able to set a target / threshold for gauges. The most common use case in my experience is to use the gauge to show %completion or how close to reaching a target a given metric is

**Comment by bondgeek** [comment]
Heatmap

**Comment by LucilleNBrown18** [comment]
Map sounds nice!

**Comment by matsstijlaart** [comment]
Boxplot!

**Comment by ceik** [comment]
Bullet Charts:

![image](https://github.com/lightdash/lightdash/assets/7529722/7f9bdc58-2a55-49d9-80e8-409ab201cc4d)

**Reply by ceik** [reply]
Essentially gauge charts for people who prefer bars over pies

**Comment by Daniel-SalazarSoplapuco** [comment]
Different X/Y Axis scales.
Logarithmic and Power would be extremely beneficial.

**Comment by Sechmail** [comment]
deck.gl geospatial charts
Any map that supports GeoJSON really

<img width="922" alt="Screenshot 2023-12-11 at 08 10 35" src="https://github.com/lightdash/lightdash/assets/3842148/17f74a48-6aca-4e3e-8e54-80da85bff070">

**Reply by alexanderwamboldt** [reply]
Yes, geospatial charting would be very useful for our work as well.

**Comment by therealchrisrock** [comment]
The Funnel Chart
![image](https://github.com/lightdash/lightdash/assets/146064866/42f494d4-e04c-45dd-8d44-ed83b7ebc61c)
![image](https://github.com/lightdash/lightdash/assets/146064866/dcff3279-e3e7-4c45-8f56-6e09b2281c56)
![image](https://github.com/lightdash/lightdash/assets/146064866/008d4362-b743-43a3-aadc-d59bc98d3470)

**Reply by Daniel-SalazarSoplapuco** [reply]
Yes I would like to add to this, this is really missing to be able to onboard the marketing department better into lightdash.

**Comment by tanlda** [comment]
Spider chart would be awesome!

![image](https://github.com/lightdash/lightdash/assets/120800972/0df4b3e8-78f4-4e5e-b269-0fe91c38c998)

**Comment by linthum** [comment]
Really need a box and wishkers plot, coming from long time Tableau user, box plot is used a lot for outlier analysis. 

<img width="730" height="320" alt="box_plot" src="https://github.com/user-attachments/assets/fba6071c-42cf-48cc-9a52-08adb851ef55" />
